Billbee.Api.Client.Test/EndPointIntegrationTests/CustomerEndPointIntegrationTest.cs

```diff
@@ -0,0 +1,212 @@
+﻿using Billbee.Api.Client.Model;
+using Billbee.Api.Client.Test.EndPointIntegrationTests.Helpers;
+
+namespace Billbee.Api.Client.Test.EndPointIntegrationTests.Helpers
+{
+    public static partial class TestData
+    {
+        public static CustomerForCreation Customer => new CustomerForCreation
+        {
+            Name = "John Doe",
+            Address = TestData.CustomerAddress,
+            Email = "john@doe.com",
+            Type = 0
+        };
+    }
+}
+
+namespace Billbee.Api.Client.Test.EndPointIntegrationTests
+{
+    [TestClass]
+    public class CustomerEndPointIntegrationTest
+    {
+#pragma warning disable CS8618
+        public TestContext TestContext { get; set; }
+#pragma warning restore CS8618
+
+        [TestInitialize]
+        public void TestInitialize()
+        {
+            IntegrationTestHelpers.CheckAccess(TestContext.ManagedType, TestContext.ManagedMethod);
+        }
+
+        [TestMethod]
+        [RequiresApiAccess]
+        public void Customer_GetCustomerList_IntegrationTest()
+        {
+            CrudHelpers.GetAll(() => IntegrationTestHelpers.ApiClient.Customer.GetCustomerList(1, int.MaxValue));
+        }
+
+        [TestMethod]
+        [RequiresApiAccess]
+        public void Customer_AddCustomer_IntegrationTest()
+        {
+            var customer = CrudHelpers.CreateApiResult(c => IntegrationTestHelpers.ApiClient.Customer.AddCustomer(c),
+                TestData.Customer);
+            Assert.AreEqual("john@doe.com", customer.Data.Email);
+            Assert.IsNotNull(customer.Data.Id);
+
+            CrudHelpers.GetOneApiResult<Customer>((id) => IntegrationTestHelpers.ApiClient.Customer.GetCustomer(id),
+                customer.Data.Id.Value);
+        }
+
+        [TestMethod]
+        [RequiresApiAccess]
+        public void Customer_AddAddressToCustomer_IntegrationTest()
+        {
+            var customer = CrudHelpers.CreateApiResult(c => IntegrationTestHelpers.ApiClient.Customer.AddCustomer(c),
+                TestData.Customer);
+
+            var customerAddress = TestData.GetCustomerAddress(customer.Data.Id);
+            var result =
+                CrudHelpers.CreateApiResult(a => IntegrationTestHelpers.ApiClient.Customer.AddAddressToCustomer(a),
+                    customerAddress);
+            Assert.IsNotNull(result.Data);
+        }
+
+        [TestMethod]
+        [RequiresApiAccess]
+        public void Customer_GetCustomerAddress_IntegrationTest()
+        {
+            var customer = CrudHelpers.CreateApiResult(c => IntegrationTestHelpers.ApiClient.Customer.AddCustomer(c),
+                TestData.Customer);
+            Assert.IsNotNull(customer);
+            Assert.IsNotNull(customer.Data);
```

Comment:
Assertion wird in CrudHelpers.CreateApiResult durchgeführt (auch im Rest der Klasse)

Reply:
Ok, wird entfernt.